src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java

```diff
@@ -126,9 +134,9 @@ protected OkHttpClient createHttpsClient() throws GeneralSecurityException {
             .connectionSpecs(singletonList(ConnectionSpec.MODERN_TLS))
             .followSslRedirects(false)
             .retryOnConnectionFailure(true)
-            .connectTimeout(5, SECONDS)
-            .readTimeout(5, SECONDS)
-            .writeTimeout(5, SECONDS)
+            .connectTimeout(keywhiz != null ? keywhiz.getConnectTimeout() : 5, SECONDS)
```

Comment:
keywhiz can't be null as we are using the baseUrl above. Also, the default value shouldn't be in the code. It should be part of the application  yaml config.

Reply:
Yeah, i will update that now.